lib/calendarium-romanum/cli.rb

```diff
@@ -17,29 +17,21 @@ def query(date_str=nil)
       end
       sanctorale = data_file.load
 
-      date =
-        if date_str
-          begin
-            Date.parse(date_str)
-          rescue ArgumentError
-            die! 'Invalid date.'
+      pcal = PerpetualCalendar.new sanctorale: sanctorale
+
+      if date_str
+        begin
+          parsed_date = DateParser.new(date_str)
+          parsed_date.payload.each do |day|
+            print_single_date(pcal, day, sanctorale)
           end
-        else
-          Date.today
+        rescue
```

Comment:
Unqualified `rescue` is bad practice. That reminds me of the fact that `DateParser` should have unit tests including error scenarios.

Reply:
Done. Created some tests for `DateParser` as well.